plugins/irma-web/index.js

```diff
@@ -16,19 +16,22 @@ module.exports = class IrmaWeb {
     );
   }
 
-  stateChange({newState, payload}) {
+  stateChange({newState, payload, universalLink}) {
     this._lastPayload = payload;
     switch(newState) {
       case 'ContinueInIrmaApp':
-        return window.setTimeout(() => this._dom.renderState(newState), 200);
+        return this._dom.renderState(newState);
       case 'ShowingQRCode':
       case 'ShowingQRCodeInstead':
-        this._dom.renderState(newState)
+        this._dom.renderState(newState);
         return QRCode.toCanvas(
           document.getElementById('irma-web-qr-canvas'),
           JSON.stringify(payload),
           {width: '230', margin: '1'}
         );
+      case 'ShowingIrmaButton':
+        this._dom.renderState(newState);
+        return document.getElementById('irma-web-button-link').setAttribute('href', universalLink);
```

Comment:
This is a DOM manipulation, so I'd rather see this in the file called `dom-manipulations.js`

Reply:
How would you propose doing that then? By adding a separate method in `dom-manipulations.js` and call that method in `index.js`? Or would you really add a if/switch kind of structure at the rend of the `renderState(state)` function and do it there? Of course the `renderPartial` must have been executed already, because otherwise the element is not available yet in the DOM.

Reply:
Hmm, I see the QR code rendering also manipulates the DOM from this switch statement. Never mind then ;)